libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -571,23 +540,53 @@ private Optional<JsonNode> generateAddressClaim(List<VerifiableCredential> vcs)
             return Optional.empty();
         }
 
-        var addressNode =
-                extractSubjectDetailFromVc(
-                        ADDRESS_PROPERTY_NAME,
-                        addressVc.get(),
-                        "Error while parsing Address CRI credential",
-                        ErrorResponse.FAILED_TO_GENERATE_ADDRESS_CLAIM);
+        var credentialSubject =
+                ((IdentityCheckCredential) addressVc.get().getCredential()).getCredentialSubject();
+
+        if (credentialSubject == null) {
+            LOGGER.error(
+                    LogHelper.buildLogMessage(
+                            ErrorResponse.CREDENTIAL_SUBJECT_MISSING.getMessage()));
+            throw new HttpResponseExceptionWithErrorBody(
+                    500, ErrorResponse.FAILED_TO_GENERATE_ADDRESS_CLAIM);
+        }
+
+        var address = credentialSubject.getAddress();
 
-        if (addressNode.isMissingNode()) {
-            LOGGER.error(LogHelper.buildLogMessage("Address property is missing from address VC"));
+        if (address == null || address.isEmpty()) {
```

Comment:
this is perfectly clear, but you can also use `software.amazon.awssdk.utils.CollectionUtils.isNullOrEmpty` if you prefer